agimus_controller_ros/agimus_controller_ros/simple_trajectory_publisher.py

```diff
@@ -124,15 +154,19 @@ def publish_mpc_input(self):
 
         if self.pin_model is None:
             self.load_models()
-
-        self.scale_amp = self.quintic_trajectory(self.t)
-        amp = self.scale_amp * self.amp
         # Currently not changing the last two joints - fingers
         # for i in range(self.pin_model.nq - 2):
+        amp, damp, ddamp = self.quint_traj.get_value_at_t(self.t)
         for i in [2, 3]:
             self.q[i] = self.q0[i] + amp * np.sin(self.w * self.t)
-            self.dq[i] = amp * self.w * np.cos(self.w * self.t)
-            self.ddq[i] = -amp * self.w * self.w * np.sin(self.w * self.t)
+            self.dq[i] = damp * np.sin(self.w * self.t) + amp * self.w * np.cos(
+                self.w * self.t
+            )
+            self.ddq[i] = (
+                ddamp * np.sin(self.w * self.t)
+                + 2 * damp * self.w * np.cos(self.w * self.t)
+                - amp * self.w * self.w * np.sin(self.w * self.t)
+            )
```

Comment:
Optional suggestions:
For math equations like this one, I usually like to introduce intermediate variable to help readability.

Here, this would be
```suggestion
            w = self.w
            sin_wt = np.sin(w * self.t)
            cos_wt = np.cos(w * self.t)
            self.dq[i] = damp * sin_wt + amp * w * cos_wt
            self.ddq[i] = (
                ddamp * sin_wt
                + 2 * damp * w * cos_wt
                - amp * w * w * sin_wt
            )
```